arcgis-ios-sdk-samples/Display information/Show a popup/ShowPopupViewController.swift

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ShowPopupViewController: UIViewController, AGSGeoViewTouchDelegate, AGSPopupsViewControllerDelegate {
+    @IBOutlet private weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+        }
+    }
+    
+    var featureLayer: AGSFeatureLayer?
+    
+    func makeMap() -> AGSMap {
+        // Create a map using a URL.
+        let mapURL = URL(string: "https://runtime.maps.arcgis.com/home/webmap/viewer.html?webmap=e4c6eb667e6c43b896691f10cc2f1580")!
+        let map = AGSMap(url: mapURL)!
+        // Load the map.
+        map.load { [weak self] (error) in
+        guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Set touch delegate.
+                self.mapView.touchDelegate = self
+                // Get the feature layer.
+                self.featureLayer = map.operationalLayers[0] as? AGSFeatureLayer
+            }
+        }
+        return map
+    }
+    
+    // MARK: - AGSGeoViewTouchDelegate
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Identify the specified feature layer.
+        mapView.identifyLayer(featureLayer!, screenPoint: screenPoint, tolerance: 12, returnPopupsOnly: false) { [weak self] (result: AGSIdentifyLayerResult) in
+            guard let self = self else { return }
+            if let error = result.error {
+                self.presentAlert(error: error)
+            } else if !result.popups.isEmpty {
+                // Unselect the previous feature and select the newest one.
+                self.featureLayer?.clearSelection()
```

Comment:
Rather than clearing the selection before selecting the next one, I would suggest clearing the selection when the popup is dismissed.

Reply:
Hm, since the popup covers the whole or most of the screen, I think it may be better to leave it as is so that the user can identify which item was selected after the popup was displayed. What do you think?

Reply:
It seems to me that the primary benefit of a selection is for the user to know for which feature the popup is being shown. I don't see as much benefit in leaving the feature selected indefinitely after the popup is dismissed, but maybe that's just me.

Reply:
I see reasons in both ways, so probably stick with the original design to keep the consistency among platforms.